python/ray/_private/runtime_env/plugin.py

```diff
@@ -158,73 +172,86 @@ def load_plugins(self, plugin_configs: List[Dict]):
                 priority = plugin_config[RAY_RUNTIME_ENV_PRIORITY_FIELD_NAME]
             else:
                 priority = plugin_class.priority
-            if (
-                not isinstance(priority, int)
-                or priority < RAY_RUNTIME_ENV_PLUGIN_MIN_PRIORITY
-                or priority > RAY_RUNTIME_ENV_PLUGIN_MAX_PRIORITY
-            ):
-                raise RuntimeError(
-                    f"Invalid runtime env priority {priority}, "
-                    "it should be an integer between "
-                    f"{RAY_RUNTIME_ENV_PLUGIN_MIN_PRIORITY} "
-                    f"and {RAY_RUNTIME_ENV_PLUGIN_MAX_PRIORITY}."
-                )
-
-            self.plugins[plugin_class.name] = RuntimeEnvPluginManager.Context(
-                plugin_class(), priority
+            self.validate_priority(priority)
+
+            class_instance = plugin_class()
+            self.plugins[plugin_class.name] = PluginSetupContext(
```

Comment:
Can we reuse the function `add_plugin` in `load_plugins`?

Reply:
That's a smart idea, let me address it in the followup PR.